src/main/java/eiteam/esteemedinnovation/engineeringtable/ContainerEngineeringTable.java

```diff
@@ -26,25 +30,25 @@
     };
 
     private TileEntityEngineeringTable tileEntity;
+    private IItemHandler itemHandler;
+    private int prevNumSlots;
+    private int numBaseSlots;
+    private boolean upgradeSlotsInitialized;
 
     public ContainerEngineeringTable(InventoryPlayer inventoryPlayer, TileEntityEngineeringTable tileEntityEngineeringTable) {
         tileEntity = tileEntityEngineeringTable;
 
-        addSlotToContainer(new Slot(tileEntityEngineeringTable, 0, 30, 35));
-        for (int i = 1; i < 10; i++) {
-            addSlotToContainer(new SlotLimitedStackSize(tileEntityEngineeringTable, i, -1000, -1000));
-        }
-        updateSlots();
-
-        int i;
+        itemHandler = tileEntityEngineeringTable.getCapability(ITEM_HANDLER_CAPABILITY, null);
+        assert itemHandler != null;
```

Comment:
You should not use assert except when testing. Change to an if statement, or an exception.